src/AutoRest.CSharp/LowLevel/AutoRest/DpgOutputLibrary.cs

```diff
@@ -23,10 +23,25 @@ internal class DpgOutputLibrary : OutputLibrary
 
         public TypeFactory TypeFactory { get; }
         public IEnumerable<EnumType> Enums => _enums.Values;
-        public IEnumerable<ModelTypeProvider> Models => _models.Values;
+        public IEnumerable<ModelTypeProvider> Models
+        {
+            get
+            {
+                foreach (var key in _models.Keys)
+                {
+                    if (key is { Namespace: "Azure.Core.Foundations", Name: "Error" })
+                    {
+                        continue;
+                    }
```

Comment:
this is not a good solution.
If we have more special types to replace, we have to change two places.
What is the issue when our previous solution?

Reply:
Isn't this our previous solution? Or are you saying we modify `_models` instead of `Models`?

Reply:
Something like:
```csharp
public IEnumerable<ModelTypeProvider> Models
{
get
{
foreach (var (key, provider) in _models)
{
var type = TypeFactory.CreateType(key);
if (type is {IsFrameworkType: false, Implemention: ModelTypeProvider m} && m == provider)
yield return m;
}
}
```
The idea here is trying to know if the type corresponding to this key has been replaced or not and we should do it in a more generic way.

Reply:
This snippet will create a lot of useless class when the type is not error. I have a new solution to resolve your first concern. You could check out the latest iteration.

Reply:
Why is this "creating" classes?
The `TypeFactory.CreateType` does not really "create" a type, it calls the `ResolveModel` method on the library in this case, which queries the `_model` dictionary using the model and returns the key.
This process never creates a class.

My initial concern is that this piece of code (both version) can only handles this "error" case, it is not generic. If we have another thing we have to replace, we need to change two places. And this is not a thing that the library should know or care about.

Reply:
So to call this `TypeFactory.Create` you have to know "If I am a model then there would be a cache". Do you think it is readable?

Anyway, what's the problem with my latest code? There are no "two places" now.

Reply:
I read your latest implementation, and there are still two places and this `Models` could only handle the replacement for errors correctly now.
This is the concern.

BTW we do not really need to consider "if this will create a new instance or not". Space consumption is the last thing we need to consider and if you notice, we are creating small object everywhere, such as when you use `typeof()` for a `CSharpType` parameter, the implicit operator always creates a new instance.

Also if you think the `TypeFactory.Create` does not have enough readability, go ahead to rename it to like `TypeFactory.GetOrCreate` to indicate it has a cache. This is not the issue we are having here.